Detectors/MUON/MID/Workflow/src/TrackerSpec.cxx

```diff
@@ -111,9 +100,30 @@ class TrackerDeviceDPL
     mNROFs += inROFRecords.size();
   }
 
+  void finaliseCCDB(o2::framework::ConcreteDataMatcher& matcher, void* obj)
+  {
+    if (o2::base::GRPGeomHelper::instance().finaliseCCDB(matcher, obj)) {
+      return;
+    }
+  }
+
  private:
+  void updateTimeDependentParams(o2::framework::ProcessingContext& pc)
+  {
+    o2::base::GRPGeomHelper::instance().checkUpdates(pc);
+
+    auto geoTrans = createTransformationFromManager(gGeoManager);
+    mTracker = std::make_unique<Tracker>(geoTrans);
+    if (!mTracker->init(mKeepAll)) {
+      LOG(error) << "Initialization of MID tracker device failed";
+    }
+    mHitMapBuilder = std::make_unique<HitMapBuilder>(geoTrans);
+  }
+
```

Comment:
Why not (re)initializing of `mTracker` and `mHitMapBuilder` in `finaliseCCDB()`, so that it is done only when the geometry is updated?